axelrod/tournament.py

```diff
@@ -307,20 +313,22 @@ def _process_done_queue(self, workers: int, done_queue: Queue,
             The number of sub-processes in existence
         done_queue : multiprocessing.Queue
             A queue containing the output dictionaries from each round robin
-        progress_bar : bool
-            Whether or not to update the tournament progress bar
         """
+        file, writer = self._get_file_objects()
```

Comment:
Does this check (the one inside `_get_file_objects` impact performance? Can this all not just be done once at the start or is that falling over with pickling?